cluster/assemblestate/selector.go

```diff
@@ -0,0 +1,380 @@
+package assemblestate
+
+import (
+	"errors"
+	"sort"
+	"time"
+
+	"github.com/snapcore/snapd/cluster/assemblestate/bimap"
+	"github.com/snapcore/snapd/cluster/assemblestate/bitset"
+	"github.com/snapcore/snapd/randutil"
+)
+
+// RouteSelector keeps track of which routes we've seen and helps pick which
+// peers to publish routes to.
+type RouteSelector interface {
+	// AddAuthoritativeRoute records a route from this local node to the given
+	// [RDT]. This route will be published to the given peer, regardless of our
+	// knowledge of that peer's identity.
```

Comment:
the doc comment and the signature don't match,  DeviceToken vs RDT? what is via?  also the text says from the local node, but then the function also take a from, should that be a to?